coordinator/core/src/main/kotlin/net/consensys/zkevm/coordinator/clients/ProverClient.kt

```diff
@@ -1,10 +1,18 @@
 package net.consensys.zkevm.coordinator.clients
 
+import net.consensys.zkevm.domain.ProofIndex
 import net.consensys.zkevm.domain.ProofToFinalize
 import net.consensys.zkevm.domain.ProofsToAggregate
 import tech.pegasys.teku.infrastructure.async.SafeFuture
 
-interface ProverClient<ProofRequest, ProofResponse> {
+interface ProverProofResponseChecker<ProofResponse> {
```

Comment:
nit: can we rename this to `ProofResponseChecker`, imo `ProverProofResponseChecker` is a bit verbose